app/flux/App.js

```diff
@@ -0,0 +1,71 @@
+// var API = require('./API');
+
+var React = require('react');
+var Router = require('react-router');
+
+var RouteHandler = Router.RouteHandler;
+
+var Footer = require('./components/Footer.react');
+var Header = require('./components/Header.react');
+
+var UserAPI = require('./API/UserAPI');
+var UserStore = require('./stores/UserStore');
+
+function getStateFromStores() {
+  return {
+    user: UserStore.user
+  }
+}
+
+class App extends React.Component {
+  constructor(props){
+    super(props);
+
+    // Only lifecycle methods receive the inheritance from React.Component.
+    this._onChange = this._onChange.bind(this);
+
+    this.state = getStateFromStores();
+  }
+
+  componentDidMount() {
+    UserStore.addChangeListener(this._onChange);
+    
+    if(!this.state.user){
+      // TODO(Bryce) - This should be a server action that emits event.
+      UserStore.isPending = true;
```

Comment:
Flux dictates that all logic should be in the action handlers, setting pending here or calling an API and waiting for a response is an anti-pattern.